File: data/derived_cot/rq3_prompting/results/translation/r1/java_to_py/code_output/PersonRequest.py
import re

class PersonRequest:
    def __init__(self, name: str, sex: str, phoneNumber: str):
        self._name = self._validate_name(name)
        self._sex = self._validate_sex(sex)
        self._phoneNumber = self._validate_phone_number(phoneNumber)

    def _validate_name(self, name: str) -> str:
        if name is None or name == "" or len(name) > 33:
            return None
        return name

    def _validate_sex(self, sex: str) -> str:
        if sex is None or sex not in ("Man", "Woman", "UGM"):
            return None
        return sex

    def _validate_phone_number(self, phoneNumber: str) -> str:
        if phoneNumber is None or len(phoneNumber) != 11 or not phoneNumber.isdigit():
            return None
        return phoneNumber

    def getName(self) -> str:
        return self._name

    def getSex(self) -> str:
        return self._sex

    def getPhoneNumber(self) -> str:
        return self._phoneNumber

File: data/derived_cot/rq3_prompting/results/translation/glm-5.1/java_to_cpp/code_output/PersonRequest.cpp
#pragma once
#include <string>
#include <optional>
#include <algorithm>
#include <cctype>

class PersonRequest {
private:
    std::optional<std::string> name;
    std::optional<std::string> sex;
    std::optional<std::string> phoneNumber;

    std::optional<std::string> validateName(const std::optional<std::string>& name) {
        if (!name.has_value() || name->empty() || name->length() > 33) {
            return std::nullopt;
        }
        return name;
    }

    std::optional<std::string> validateSex(const std::optional<std::string>& sex) {
        if (!sex.has_value() || (*sex != "Man" && *sex != "Woman" && *sex != "UGM")) {
            return std::nullopt;
        }
        return sex;
    }

    std::optional<std::string> validatePhoneNumber(const std::optional<std::string>& phoneNumber) {
        if (!phoneNumber.has_value() || phoneNumber->empty() || phoneNumber->length() != 11 ||
            !std::all_of(phoneNumber->begin(), phoneNumber->end(),
                         [](unsigned char c) { return std::isdigit(c); })) {
            return std::nullopt;
        }
        return phoneNumber;
    }

public:
    PersonRequest(const std::optional<std::string>& name,
                  const std::optional<std::string>& sex,
                  const std::optional<std::string>& phoneNumber) {
        this->name = validateName(name);
        this->sex = validateSex(sex);
        this->phoneNumber = validatePhoneNumber(phoneNumber);
    }

    std::optional<std::string> getName() const {
        return name;
    }

    std::optional<std::string> getSex() const {
        return sex;
    }

    std::optional<std::string> getPhoneNumber() const {
        return phoneNumber;
    }
};